clang/test/OpenMP/assume_lambda.cpp

```diff
@@ -0,0 +1,31 @@
+// RUN: %clang_cc1 -fopenmp -x c++ -std=c++11 -ast-print %s | FileCheck %s
+// expected-no-diagnostics
+
+extern int bar(int);
+
+int foo(int arg)
+{
+  #pragma omp assume no_openmp_routines
+  {
+    auto fn = [](int x) { return bar(x); };
+// CHECK: auto fn = [](int x) {
+    return fn(5);
+  }
+}
+
+class C {
+public:
+  int foo(int a);
+};
+
+// We're really just checking that this parses.  All the assumptions are thrown
+// away immediately for now.
+int C::foo(int a)
+{
+  #pragma omp assume holds(sizeof(T) == 8) absent(parallel)
+  {
+    auto fn = [](int x) { return bar(x); };
+// CHECK: auto fn = [](int x) {
+    return fn(5);
+  }
+}
```

Comment:
empty line EoF

Reply:
Ugh, a bad habit of that editor. I guess I should try to figure out a way of stopping it doing that automatically...

Reply:
I mean, we need an empty line at the end of file.

Reply:
Understood! There is indeed a vscode option to add the missing newline (it appears to be turned off by default for some bizarre reason). I'll push a new version with them in.